src/commands/Generation.ts

```diff
@@ -0,0 +1,53 @@
+import { Client, Message, TextChannel } from "discord.js";
+import { ComplimentGenerator, InsultGenerator, JokeGenerator } from "../generators";
+import { CommandType, ICommand } from "./Command";
+
+/*
+  Extension of the base Command interface for the Generate command.
+*/
+export interface IGenerationCommand extends ICommand {
+  // This is the channel ID to explicitly send the message to.
+  channelIDToSendMessage: string;
+  message: string;
+}
+
+/*
+  Implementation of Impersonation command execution.
+*/
+export class GenerationCommand implements IGenerationCommand {
+  type: CommandType
+  discordMessage?: Message
+  channelIDToSendMessage: string;
+  message: string
+  constructor(channelIDToSendMessage: string, message: string){
+    this.channelIDToSendMessage = channelIDToSendMessage
```

Comment:
There is no need for GenerationCommand to have the explicit channel ID. The response should be sent to the same channel the original message is from.

Reply:
done